interface-definitions/include/firewall/gre.xml.i

```diff
@@ -0,0 +1,126 @@
+<!-- include start from firewall/gre.xml.i -->
+<node name="gre">
+  <properties>
+    <help>GRE fields to match</help>
+  </properties>
+  <children>
+    <node name="flags">
+      <properties>
+        <help>GRE flag bits to match</help>
+      </properties>
+      <children>
+        <leafNode name="key">
+          <properties>
+            <help>Header includes optional key field</help>
+            <valueless/>
+          </properties>
+        </leafNode>
+        <leafNode name="checksum">
+          <properties>
+            <help>Header includes optional checksum</help>
+            <valueless/>
+          </properties>
+        </leafNode>
+        <leafNode name="sequence">
+          <properties>
+            <help>Header includes a sequence number field</help>
+            <valueless/>
+          </properties>
+        </leafNode>
+        <node name="not">
+          <properties>
+            <help>Match flags not set</help>
+          </properties>
+          <children>
+            <leafNode name="key">
+              <properties>
+                <help>Header does not include optional key field</help>
+                <valueless/>
+              </properties>
+            </leafNode>
+            <leafNode name="checksum">
+              <properties>
+                <help>Header does not include optional checksum</help>
+                <valueless/>
+              </properties>
+            </leafNode>
+            <leafNode name="sequence">
+              <properties>
+                <help>Header does not include a sequence number field</help>
+                <valueless/>
+              </properties>
+            </leafNode>
+          </children>
+        </node>
+      </children>
+    </node>
+    <leafNode name="inner-proto">
+      <properties>
+        <help>EtherType of encapsulated packet</help>
+        <completionHelp>
+          <list>ip ip6 arp vlan 8021q 8021ad</list>
+        </completionHelp>
+        <valueHelp>
+          <format>u32:0-65535</format>
+          <description>Ethernet protocol number</description>
+        </valueHelp>
+        <valueHelp>
+          <format>ip</format>
+          <description>IPv4</description>
+        </valueHelp>
+        <valueHelp>
+          <format>ip6</format>
+          <description>IPv6</description>
+        </valueHelp>
+        <valueHelp>
+          <format>arp</format>
+          <description>Address Resolution Protocol</description>
+        </valueHelp>
+        <valueHelp>
+          <format>vlan</format>
+          <description>VLAN-tagged Ethernet</description>
+        </valueHelp>
+        <valueHelp>
+          <format>8021q</format>
+          <description>VLAN-tagged Ethernet</description>
+        </valueHelp>
+        <valueHelp>
+          <format>8021ad</format>
+          <description>Bridged Ethernet</description>
```

Comment:
I'm updating it to track the same terminology as include/vlan-protocol.xml.i (slightly less verbose): "Provider Bridging (IEEE 802.1ad, Q-in-Q)". 

If you think it's useful to mention VLAN stacking as well when stuffing QinQ into GRE, it's no big effort for me to tweak again. 

Since it's being tweaked, I'll add an alias for gretap 0x6558 as well, as it's an explicitly supported GRE interface type in VyOS.